elastic/terraform-provider-ec
# Feature/530/migrate to plugin framework

## Description


Migration to [TF Plugin Framework](https://developer.hashicorp.com/terraform/plugin/framework).
New schema for `ec_deployment`.

*Limitations*
* State upgrade is not currently implemented for `ec_deployment`. The recommended way to port existing resources is importing.
* The migration is based on the provider version 0.4.1. We plan to bring changes from 0.5.0 in future releases.
* Few acceptance tests fail. Please see the testing section below.

## Related Issues






## Motivation and Context


## How Has This Been Tested?

* unit tests (majority of the relevant unit tests were migrated)
* acceptance tests
* manual testing

There are still few acceptance tests that fail:
- 4 new acceptance tests (their names mention version 0.4.1) that verify migration of other data sources and resources from 0.4.1 to 0.6.0. To fix them we need implement a state upgrade for `ec_deployment`.
- `TestAccDeploymentExtension_pluginDownload` fails on 0.4.1 with the same error ("Custom plugin is not allowed at your subscription level.")
- `TestAccDeploymentExtension_bundleFile` fails quite often on both 0.4.1 and the branch with the "failed to upload file" error.
- `TestAccDeployment_withExtension` fails quite often with the same error as `TestAccDeploymentExtension_bundleFile`.





## Types of Changes


- [ ] Refactoring (improves code quality but has no user-facing effect)

## Readiness Checklist
